plugins/modules/sns_topic.py

```diff
@@ -401,6 +410,36 @@ def _set_topic_subs(self):
                     self.module.fail_json_aws(e, msg="Couldn't subscribe to topic %s" % self.topic_arn)
         return changed
 
+    def _set_topic_subs_attributes(self):
+        changed = False
+        desired_attributes = dict()
+        for sub in self.subscriptions:
+            sub_key = (sub['protocol'], self._canonicalize_endpoint(sub['protocol'], sub['endpoint']))
+            desired_attributes[sub_key] = sub.get('attributes', {})
+
+        for sub in self._list_topic_subscriptions():
+            sub_key = (sub['Protocol'], sub['Endpoint'])
+            sub_arn = sub['SubscriptionArn']
+            try:
+                sub_current_attributes = self.connection.get_subscription_attributes(SubscriptionArn=sub_arn)['Attributes']
+            except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+                self.module.fail_json_aws(e, "Couldn't get subscription attributes for subscription %s" % sub_arn)
+
+            raw_message = desired_attributes[sub_key].get('RawMessageDelivery')
+            if raw_message is not None:
+                if sub_current_attributes['RawMessageDelivery'] != raw_message:
+                    changed = True
+                    self.subscriptions_attributes_set.append((sub_arn, 'RawMessageDelivery'))
+                    if not self.check_mode:
+                        try:
+                            self.connection.set_subscription_attributes(SubscriptionArn=sub_arn,
+                                                                        AttributeName='RawMessageDelivery',
+                                                                        AttributeValue=raw_message)
```

Comment:
Regrettably, AWS expects this value to be a string of true or false, not an actual bool.  Uncovered this while testing your terminator PR.
```
    "msg": "Couldn't set RawMessageDelivery subscription attribute: Parameter validation failed:\nInvalid type for parameter AttributeValue, value: True, type: <class 'bool'>, valid types: <class 'str'>",
```
https://docs.aws.amazon.com/sns/latest/api/API_SetSubscriptionAttributes.html

Reply:
Wasn't aware of that. Pushed a fix to convert all attribute values to strings before assigning them to desired_attributes dict.